### PR TITLE
Remove unnecessary checks in Hexen

### PR DESCRIPTION
**Pull request by mfrancis95**

### CODE REVIEW

File: src/hexen/p_inter.c [hunk 1]
```diff
@@ -2174,7 +2174,7 @@ void P_PoisonDamage(player_t * player, mobj_t * source, int damage,
     {                           // mobj is invulnerable
         return;
     }
-    if (player && gameskill == sk_baby)
```
Comment (turol) [inline src/hexen/p_inter.c]: Is this _really_ safe? Are the any callers of `P_PoisonDamage` where `player` is null? Should the check simply be moved earlier?

Comment (fabiangreffrath) [reply src/hexen/p_inter.c]: `player` is already dereferenced 11 lines up.

Comment (turol) [reply src/hexen/p_inter.c]: But how do we know _that_ is safe? Has anyone actually checked all the callers?

Comment (mfrancis95) [reply src/hexen/p_inter.c]: On [this line](https://github.com/chocolate-doom/chocolate-doom/blob/51e202b5b2495a1f25e636eb44cdc310801ba311/src/hexen/p_inter.c#L1890), `P_PoisonDamage` will only get called if `target->player` is not null, therefore from the perspective of the `P_PoisonDamage` function, `player` will always be non-null.

Comment (mfrancis95) [reply src/hexen/p_inter.c]: The only other area where `P_PoisonDamage` is getting called is from within `P_PlayerThink, where `player` is being dereferenced countless times and is certainly not null.

Comment (turol) [reply src/hexen/p_inter.c]: Ok, add those comments to the git commit messages and I'll approve this.

**Comment by mfrancis95** [comment]
@turol

**Comment by turol** [comment]
For the future, it's git convention to have a short log message which fits on the first line of the message and then a longer message after that. Only the short one is shown by default, making the list of changes more readable.

**Comment by turol** [comment]
@mfrancis95 Do you want to push merge or should I?

**Pull request merged**